src/library/scala/collection/Iterator.scala

```diff
@@ -262,14 +256,13 @@ trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Ite
       else go(step)
     }
 
-    def hasNext = filled || fill()
+    def hasNext: Boolean =
+      filled || fill()
+
     @throws[NoSuchElementException]
     def next(): immutable.Seq[B] = {
-      if (!filled)
-        fill()
-
-      if (!filled)
-        throw new NoSuchElementException("next on empty iterator")
+      if (!filled) fill()
+      if (!filled) throw new NoSuchElementException("next on empty iterator")
```

Comment:
if we're already editing this, `Iterator.empty.next()` would be the idiomatic way of doing this (it obviates the need to keep the exception message consistent across lots of implementations)

Reply:
it's a one-liner so the line is filled.